components/login/login_controller/login_controller.jsx

```diff
@@ -364,11 +364,13 @@ class LoginController extends React.Component {
         if (this.props.enableCustomBrand) {
             const text = this.props.customBrandText || '';
             const formattedText = TextFormatting.formatText(text);
+            const brandImageUrl = Client4.getBrandImageUrl(0);
 
             return (
                 <div>
-                    <img
-                        src={Client4.getBrandImageUrl(0)}
+                    <object
```

Comment:
@hmhealey @deanwhillier I'm not 100% sure on the browser-compatibility of using an object tag here, but my research suggests it is compatible back to IE8. Object data fallbacks should be acceptable in general [according to W3C](https://www.w3.org/TR/WCAG20-TECHS/H53.html)

Reply:
It's a creative option for sure!

My only concern is that the `<object>` tag unfortunately has no semantic meaning. I personally prefer maintaining semantic meaning where possible (3/5). What about a React equivalent of something like this?

```
<img id="image" src="https://community.mattermost.com/non-existent-image.jpg" onerror="handleError()" />
<script>
        function handleError() {
                document.getElementById("image").style.display = "none";
        }
</script>
```

I'm not yet sure what's supposed to happen with the custom branding. Should there be a fallback Mattermost logo in the event that a custom logo isn't provided? If so, adding some CSS and/or JS to the above could render the MM logo if a custom logo failed to render. Or perhaps a text fallback of the company's name could be used when a logo fails to load?

Thoughts?

Reply:
Of course, a proper fix would be to have the backend ensure it only sends a reference to a valid image. :D

Reply:
@deanwhillier In terms of behavior, I thought it was specified in the ticket that no image should appear, but that conversation may have occurred in another channel. I can get clarification here from @esethna . I also was thinking the Mattermost logo should appear by default.

In terms of semantics, I agree, however I'm not sure about adding additional code for handling a broken image that should be fixed on the API. But, the benefit is that it I expect it would eliminate the 404 error in the console. Let me give that a try.

Reply:
Apologies @bradjcoughlin, the ticket does say the area should remain blank if no image is available.

Reply:
@deanwhillier it was "hidden" under the QA instructions

Looks like there is more work involved to handling a broken image link in React. For this issue (presuming it is an edge case) I'm in favor of using the object tag given that the root cause should be addressed on the server side. I can file a bug to fix this in the admin console and/or in the API response.

Reply:
Sound good. Maybe also reference changing this back to an `<img>` tag in the ticket when it is fixed in the appropriate spot?

Reply:
Generally, we would use an `img` tag with an `onError` in cases like this or we would have some prop that tells the client whether or not to even show that. I'm fine with continuing to use an `object` tag here though since it also works

Reply:
After further investigation I think we should avoid the object tag. 1) The issue of semantics, and 2) the fact that we can't be sure of the file type (jpg or png). My proposal is to add the error handing until the backend issue is resolved by [MM-13926](https://mattermost.atlassian.net/browse/MM-13926)